src/App.js

```diff
@@ -1,39 +1,145 @@
 import React from 'react';
-import TodoList from './components/TodoComponents/TodoList'
-// import Todo from './components/TodoComponents/Todo'
+import './App.css';
 import TodoForm from './components/TodoComponents/TodoForm'
+import Todo from './components/TodoComponents/Todo'
 
 class App extends React.Component {
+  inputElement = React.createRef()  
   constructor() {
     super();
     this.state = {
       listArray: [],
-    };
+      currentItem: {text:'', key:'', completed:''}
+    }
   }
 
-  addListItem = newItem => {
-    this.setState({ listArray: [...this.state.listArray, newItem] })
+  clearCompleted = e => {
+    e.preventDefault()
+    this.setState({
+      listArray: this.state.listArray.filter(current => {
+        return !current.completed
+      })
+    })
+  }
+
+  completeItem = task => {
+    const completedItems = this.state.listArray.map(item => {
+      if (item.key === task.key) {
+        return { 
+          ...task, 
+          completed: !task.completed
+        }
+      } else return item
+    })
+    this.setState({
+      listArray: completedItems
+    })
   }
 
-  clearAllItems() {
+  handleInput = e => {
+    const itemText = e.target.value
+    const currentItem = { text: itemText, key: Date.now(), completed: false }
     this.setState({
-      emptyArray: []
+      currentItem,
     })
   }
 
+  addItem = e => {
+    e.preventDefault()
+    const newItem = this.state.currentItem
+    if (newItem.text !== '') {
+      const items = [...this.state.listArray, newItem]
+      this.setState({
+        listArray: items,
```

Comment:
Nice touch with the `if` statement for preventing the submission of empty todo items!